File: src/ua/nure/lab2/PotholerNode.java
package ua.nure.lab2;

import java.util.ArrayList;
import java.util.List;

public class PotholerNode {

    private int x;
    private int y;
    private boolean pit;
    private boolean vampus;
    private boolean isVampusAlive;
    private boolean gold;
    private List<PotholerAgent> hunters;

    public PotholerNode(int x, int y) {
        this.x = x;
        this.y = y;
        this.pit = false;
        this.vampus = false;
        this.isVampusAlive = false;
        this.gold = false;
        this.hunters = new ArrayList<>();
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public boolean isPit() {
        return pit;
    }

    public void setPit(boolean pit) {
        this.pit = pit;
    }

    public boolean isVampus() {
        return vampus;
    }

    public void setVampus(boolean vampus) {
        this.vampus = vampus;
    }

    public boolean isVampusAlive() {
        return isVampusAlive;
    }

    public void setVampusAlive(boolean vampusAlive) {
        isVampusAlive = vampusAlive;
    }

    public boolean isGold() {
        return gold;
    }

    public void setGold(boolean gold) {
        this.gold = gold;
    }

    public List<PotholerAgent> getHunters() {
        return hunters;
    }

    public void setHunters(List<PotholerAgent> hunters) {
        this.hunters = hunters;
    }
}

File: MSiT_Lab3/src/main/scala/wumpus/PotholerNode.scala
package wumpus

import java.util

class PotholerNode {
  private var x = 0
  private var y = 0
  private var pit = false
  private var vampus = false
  private var isVampusAlive = false
  private var gold = false
  private var hunters = null

  def this(x: Int, y: Int) {
    this()
    this.x = x
    this.y = y
    this.pit = false
    this.vampus = false
    this.isVampusAlive = false
    this.gold = false
    this.hunters = new util.ArrayList[PotholerAgent]
  }

  def getX: Int = x

  def setX(x: Int): Unit = {
    this.x = x
  }

  def getY: Int = y

  def setY(y: Int): Unit = {
    this.y = y
  }

  def isPit: Boolean = pit

  def setPit(pit: Boolean): Unit = {
    this.pit = pit
  }

  def isVampus: Boolean = vampus

  def setVampus(vampus: Boolean): Unit = {
    this.vampus = vampus
  }

  def isVampusAlive: Boolean = isVampusAlive

  def setVampusAlive(vampusAlive: Boolean): Unit = {
    isVampusAlive = vampusAlive
  }

  def isGold: Boolean = gold

  def setGold(gold: Boolean): Unit = {
    this.gold = gold
  }

  def getHunters: util.List[PotholerAgent] = hunters

  def setHunters(hunters: util.List[PotholerAgent]): Unit = {
    this.hunters = hunters
  }

}
